lib/SIL/Utils/OwnershipUtils.cpp

```diff
@@ -83,6 +83,23 @@ bool swift::hasPointerEscape(BorrowedValue value) {
   return false;
 }
 
+bool swift::hasPointerEscape(SILValue value) {
+  if (auto borrowedValue = BorrowedValue(value)) {
+    return hasPointerEscape(borrowedValue);
+  }
+  assert(value->getOwnershipKind() == OwnershipKind::Owned);
+  for (auto use : value->getUses()) {
```

Comment:
Thank you, great point.  From recent discussion with @atrick, my understanding was that a ForwardingConsumes shouldn't be considered part of the same lifetime for the purposes of pointer escapes.  If that applies even to branches, since a `br(%owned)` is a forwarding consume of `%owned`, it wouldn't be included.  On the other hand, excluding phi-extended lifetime is inconsistent with the intuition that "things should be the same" whether (1) there's a merge and a phi or (2) the function body portion is cloned for each incoming value.  So it's seeming to me like we do need to look through phis like you say.

If we need to look through phis, phis also probably need to be rejected as inputs to this function like they are rejected by `hasPointerEscape(BorrowedValue)`.  It seems we also might not want to count them as "owned value introducers" (or allow them to be used in @atrick's forthcoming `OwnedValue` abstraction).

I updated the patch to look through phis and ban them as input.